cedar-policy-validator/src/typecheck.rs

```diff
@@ -155,7 +155,10 @@ impl<'a> Typechecker<'a> {
                 (false, true, None) => PolicyCheck::Fail(type_errors),
                 (false, true, Some(e)) => PolicyCheck::Success(e),
                 (false, false, _) => PolicyCheck::Fail(type_errors),
-                (true, _, _) => PolicyCheck::Irrelevant(type_errors),
+                (true, _, Some(e)) => PolicyCheck::Irrelevant(type_errors, e),
+                // PANIC SAFETY: `is_false` implies e is `Some(Lit(Bool(false)))`
+                #[allow(clippy::unreachable)]
+                (true, _, None) => unreachable!(),
```

Comment:
This and the one below should be safe, but the panic safety comment is too strong. `ty.typechecked() == true` implies `ty.into_typed_expr().is_some()`, but it might not be a lit bool

Reply:
Changed to "`is_false` implies `e` has a type implies `Some(e)`."